app/src/main/java/com/example/poste/activities/LoginActivity.java

```diff
@@ -1,99 +1,127 @@
 
-        //Page Navigation starts
-        buttonLoginSubmit = findViewById(R.id.LoginLoginbtn);
+        buttonLoginSubmit = findViewById(R.id.loginLoginbtn);
 
-        //Creating the link to Dashboard page
-        buttonLoginSubmit.setOnClickListener(view -> checkLogin());
-        //Page Navigation ends
-    }
+        /**
+         * adds a hyperlink to redirect user to the login page
+         */
+        SpannableString spannable = new SpannableString(hyperlinkTextView.getText());
+        ClickableSpan clickableSpan = new ClickableSpan() {
+            @Override
+            public void onClick(View widget) {
+                // Start the SecondActivity when the link is clicked
+                Intent intent = new Intent(LoginActivity.this, RegisterActivity.class);
+                startActivity(intent);
+            }
+        };
 
-    void navigateToDashboard(){
-        finish();
-        Intent intent = new Intent(LoginActivity.this, DashboardActivity.class);
-        startActivity(intent);
-    }
+        spannable.setSpan(clickableSpan, 0, hyperlinkTextView.getText().length(), SpannableString.SPAN_EXCLUSIVE_EXCLUSIVE);
 
+        hyperlinkTextView.setText(spannable);
+        hyperlinkTextView.setMovementMethod(LinkMovementMethod.getInstance());
 
-    /**
-     * Triggers when LOGIN Button clicked from LOGIN page.
-     * Parses email and password from text fields, and sends an API call to validate user.
-     * If user is validated, sets as current user and return to dashboard.
-     * Otherwise, display error.
-     */
-    public void checkLogin() {
-        email = usernameField.getText().toString();
-        password = passwordField.getText().toString();
-        if(email.isEmpty() || password.isEmpty())
-            Toast.makeText(this, "Please enter your login details", Toast.LENGTH_LONG).show();
-        else {
-            try {
-                User user = API.login(email, password);
-                if (user != null) {
-                    Toast.makeText(LoginActivity.this, getString(R.string.login_successful), Toast.LENGTH_LONG).show();
-                    PosteApplication.setCurrentUser(loginUser);
-                    navigateToDashboard();
-                } else {
-                    Toast.makeText(LoginActivity.this, getString(R.string.login_invalid_credentials), Toast.LENGTH_LONG).show();
-                }
-            } catch (APIException e) {
-                e.printStackTrace();
-                Toast.makeText(LoginActivity.this, getString(R.string.internal_error), Toast.LENGTH_LONG).show();
+        /**
+         * Function that runs when login button is pressed
+         * Makes a request to the API and gets a response in return
+         */
+        buttonLoginSubmit.setOnClickListener(view -> {
+            String email = usernameField.getText().toString();
+            String password = passwordField.getText().toString();
+            if (!validateEmail(LoginActivity.this, email)){
```

Comment:
Might be useful to also call validatePassword(), so we can enforce some password validity client-side

Reply:
The current method for validatePassword required 2 string sof password and confirm, so I created a new method for it.